music_assistant/providers/spotify/streaming.py

```diff
@@ -0,0 +1,357 @@
+            args.extend(["--start-position", str(int(seek_position))])
+
+        return args
+
+    async def _stream_track_via_librespot(
+        self, streamdetails: StreamDetails, seek_position: int = 0
+    ) -> AsyncGenerator[bytes, None]:
+        """Stream track using librespot."""
+        spotify_uri = f"spotify://track:{streamdetails.item_id}"
+        self.logger.log(VERBOSE_LOG_LEVEL, f"Start streaming {spotify_uri} using librespot")
+        self.logger.info(f"Starting librespot for track: {spotify_uri}")
+
+        for attempt in (1, 2):
+            args = self._get_librespot_args(spotify_uri, "track", seek_position)
+            self.logger.debug(f"Librespot command: {' '.join(args)}")
+
+            async with AsyncProcess(
+                args,
+                stdout=True,
+                stderr=None if self.logger.isEnabledFor(VERBOSE_LOG_LEVEL) else False,
+                name="librespot",
+            ) as librespot_proc:
+                chunks_received = 0
+
+                try:
+                    chunk = await asyncio.wait_for(librespot_proc.read(64000), timeout=5 * attempt)
+
+                    if not chunk:
+                        raise AudioError("No audio received from librespot - empty chunk")
+
+                    self.logger.debug(
+                        f"Successfully received initial audio chunk ({len(chunk)} bytes) "
+                        f"on attempt {attempt}"
+                    )
+                    yield chunk
+                    chunks_received = 1
+
+                    async for chunk in librespot_proc.iter_chunked():
+                        chunks_received += 1
+                        yield chunk
+
+                    self.logger.debug(
+                        f"Completed streaming track - total chunks: {chunks_received}"
+                    )
+                    return
+
+                except (TimeoutError, AudioError):
+                    if chunks_received > 50:
+                        self.logger.warning(
+                            f"Stream interrupted after receiving {chunks_received} chunks - "
+                            f"treating as successful completion"
+                        )
+                        return
+
+                    err_msg = "No audio received from librespot within timeout"
+
+                    if attempt == 2:
+                        self.logger.error(f"All attempts failed for track {streamdetails.item_id}")
+                        raise AudioError(err_msg)
+                    else:
+                        self.logger.warning(f"{err_msg} - will retry once")
+                        continue
+
+    async def _stream_episode_via_librespot(
+        self, streamdetails: StreamDetails, seek_position: int = 0
+    ) -> AsyncGenerator[bytes, None]:
+        """Stream episode using librespot with retry logic."""
+        self.logger.info(f"Episode {streamdetails.item_id}")
+        spotify_uri = f"spotify:episode:{streamdetails.item_id}"
+

```

Comment:
This is a lot of duplicated code. Other than the media type (and thus the uri) there is no difference between streaming a track and an episode. So this should be combined into one function